com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/eval/JdtEvaluationProvider.java

```diff
@@ -178,6 +188,37 @@ public void clearState(ThreadReference thread) {
         }
     }
 
+    private boolean ensureDebugTarget(VirtualMachine vm) {
+        String projectName = (String) options.get(Constants.PROJECTNAME);
+        if (debugTarget == null) {
+            if (project == null) {
+                if (StringUtils.isBlank(projectName)) {
+                    logger.severe("Cannot evaluate when project is not specified.");
+                    throw new IllegalStateException("Please specify projectName in launch.json.");
+                }
+                project = JdtUtils.getJavaProject(projectName);
+            }
+
+            if (project == null) {
+                new IllegalStateException(String.format("Project %s cannot be found.", projectName));
+                return false;
+            }
+            if (launch == null) {
+                launch = createILaunchMock(project);
+            }
+        }
+
+        if (debugTarget == null) {
```

Comment:
The outside is already have this logic

```java
debugTarget == null
```

Reply:
Why do we have two if blocks with the same condition to check whether debugTarget is null?

Reply:
Old issue in previous code, will fix it.